### PR TITLE
Completed basic-UI

### PR DESCRIPTION
**Pull request by qburst-shanavas**
@madhukarmmallia @AmalQB @sam-qburst @abhilashsajeev

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,15 +16,20 @@ var app = express();
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'jade');
 
+app.use(express.static('views'));
+
 // uncomment after placing your favicon in /public
 //app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
```
Comment (abhilashsajeev) [inline app.js]: Remove commented  code

File: app.js [hunk 2]
```diff
@@ -16,15 +16,20 @@ var app = express();
...
+app.use(bodyParser.json({
+  limit: '500mb'
+}));
 app.use(bodyParser.urlencoded({
+  limit: '500mb',
   extended: false
 }));
 app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
+//app.use(express.static(path.join(__dirname, 'public')));
```
Comment (abhilashsajeev) [inline app.js]: remove

File: views/s3uploader.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+var app =  angular.module('s3uploader', []);
+app.controller('fileSelection', function ($http){
+		var fileName;
+		var file;
+		this.fileNameFetch=function() {
+			fileName = document.getElementById('myFile');
```
Comment (abhilashsajeev) [inline views/s3uploader.js]: What indentation level are you keeping? 2 or 4

File: package.json [hunk 4]
```diff
@@ -13,9 +13,11 @@
     "csvtojson": "^0.5.11",
     "debug": "~2.2.0",
     "express": "~4.13.1",
+    "jade": "^1.11.0",
```
Comment (abhilashsajeev) [inline package.json]: why jade again?

File: routes/index.js [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
 /* GET home page. */
-router.get('/', function (req, res, next) {
-  res.render('index', {
-    title: 'Express'
+router.post('/api/upload', upload.single('file'), function (req, res, next) {
+  console.log(req.file.path);
+  fs.readFile(req.file.path, function (err, contents) {
+    console.log(contents);
   });
+  //res.send(success);
```
Comment (abhilashsajeev) [inline routes/index.js]: remove commented code

File: routes/index.js [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+var upload = multer({
+  dest: 'uploads/'
+});
 
 /* GET home page. */
-router.get('/', function (req, res, next) {
-  res.render('index', {
-    title: 'Express'
+router.post('/api/upload', upload.single('file'), function (req, res, next) {
+  console.log(req.file.path);
```
Comment (madhukarmmallia) [inline routes/index.js]: Remove unnecessary console.log.

File: routes/index.js [hunk 7]
```diff
@@ -1,12 +1,19 @@
 'use strict';
 var express = require('express');
 var router = express.Router();
+var fs = require('fs');
+var multer = require('multer');
+var upload = multer({
+  dest: 'uploads/'
```
Comment (madhukarmmallia) [inline routes/index.js]: Would be great if you could move this to a configuration file.

File: routes/index.js [hunk 8]
```diff
@@ -1,12 +1,19 @@
...
+});
 
 /* GET home page. */
-router.get('/', function (req, res, next) {
-  res.render('index', {
-    title: 'Express'
+router.post('/api/upload', upload.single('file'), function (req, res, next) {
+  console.log(req.file.path);
+  fs.readFile(req.file.path, function (err, contents) {
+    console.log(contents);
```
Comment (madhukarmmallia) [inline routes/index.js]: Is this logging required?

File: views/s3uploader.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var app = angular.module('s3uploader', []);
+app.controller('fileSelection', function ($http) {
+  var fileName;
+  var file;
+  this.fileNameFetch = function () {
+    fileName = document.getElementById('myFile');
+    console.log(fileName.value);
```
Comment (madhukarmmallia) [inline views/s3uploader.js]: Again, is this console required?

File: views/s3uploader.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var app = angular.module('s3uploader', []);
+app.controller('fileSelection', function ($http) {
+  var fileName;
+  var file;
+  this.fileNameFetch = function () {
+    fileName = document.getElementById('myFile');
```
Comment (madhukarmmallia) [inline views/s3uploader.js]: Stick to angular if you are using that. This is not the angular way of uploading file.

File: views/index.html [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  	<h1>S3 uploader Demo</h1>
+  	<div>
+  		<form ng-submit="selection.fileNameFetch()">
+	  		<input type="file" size="60" id="myFile" name="pic" accept=".xlsx">
+	  		<input type="submit">
+		</form>
+	</div>  	
+</body>
+
+</html>
```
Comment (madhukarmmallia) [inline views/index.html]: Reduce indentation. Load the form as a template.

File: views/s3uploader.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    console.log(fileName.value);
+    file = fileName.files[0];
+    var fd = new FormData();
+    fd.append('file', file);
+    $http.post('/api/upload', fd, {
```
Comment (madhukarmmallia) [inline views/s3uploader.js]: Move this to a service.

File: views/index.html [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html ng-app = "s3uploader">
+<head>
+	<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular.min.js"></script>
+	<script src="s3uploader.js"></script>
```
Comment (madhukarmmallia) [inline views/index.html]: Use gulp to load the AngularJS app files.

**Comment by madhukarmmallia** [comment]
@shanavasHassan Please check the folder structure for frontend files. You don't keep a controller under the folder `views`. Please go through the document for the coding standards, folder structure, file naming convention which we use. Stick to that. Make the changes and let us know.

File: app.js [hunk 14]
```diff
@@ -12,18 +12,18 @@ var users = require('./routes/users');
...
 
 // uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
 app.use(logger('dev'));
-app.use(bodyParser.json());
+app.use(bodyParser.json({
+  limit: '500mb'
+}));
 app.use(bodyParser.urlencoded({
+  limit: '500mb',
```
Comment (abhilashsajeev) [inline app.js:22]: Try to use this in config

File: gulpfile.js [hunk 15]
```diff
@@ -1,16 +1,42 @@
 'use strict';
 
 var gulp = require('gulp');
+var nodemon = require('gulp-nodemon');
+var uglify = require('gulp-uglify');
+var concat = require('gulp-concat');
 
 gulp.paths = {
   scripts: ['routes/*.js',
```
Comment (abhilashsajeev) [inline gulpfile.js:9]: 'routes/*.js ' in new line

File: public/angular-ui-router.min.js [hunk 16]
```diff
@@ -0,0 +1,8 @@
+/**
```
Comment (abhilashsajeev) [inline public/angular-ui-router.min.js:1]: Use bower to manage angular dependencies. Use bower.json. Do not commit angular files to github repo.

File: public/file-upload/file-upload.html [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+  <p class="container text-center">
+  	<input type="file" file-model="myFile" accept=".xlsx">
+  </p>
+  <p>
+    <button class="btn btn-default btn-primary" ng-click="uploadFile()" ng-disabled="!myFile">upload me</button>	
+	</p>
+	<div class="alert alert-success" role="alert" ng-show="alertShow">
+  <a href="#" class="alert-link">File uploaded successfully!!</a>
+</div>
+</div>
```
Comment (abhilashsajeev) [inline public/file-upload/file-upload.html:12]: Trailing space  missing

File: public/router.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+var s3uploader = angular.module('s3Uploader', ['ui.router'])
+  .config(['$stateProvider', '$urlRouterProvider', function ($stateProvider, $urlRouterProvider) {
+    $urlRouterProvider.otherwise('/');
+    $stateProvider
```
Comment (abhilashsajeev) [inline public/router.js:6]: What is the busineess with code in 'public/file-upload/file-upload.js'  both seem to look same?

File: public/source-integrated-file.js [hunk 19]
```diff
@@ -0,0 +1 @@
+"use strict";var s3uploader=angular.module("s3Uploader",["ui.router"]).config(["$stateProvider","$urlRouterProvider",function(e,o){o.otherwise("/"),e.state("home",{url:"/",templateUrl:"/file-upload/file-upload.html",controller:"fileSelection as fileupCtrl"})}]);s3uploader.controller("fileSelection",["$scope","fileUpload",function(e,o){e.alertShow=!1,e.uploadFile=function(){var l=function(){e.alertShow=!0,setTimeout(function(){e.alertShow=!1,e.$apply()},2e3)},i=e.myFile;if(i){var t="/api/upload";o.uploadFileToUrl(i,t,l)}}}]),s3uploader.directive("fileModel",["$parse",function(e){return{restrict:"A",link:function(o,l,i){var t=e(i.fileModel).assign;l.bind("change",function(){o.$apply(function(){t(o,l[0].files[0])})})}}}]),s3uploader.service("fileUpload",["$http",function(e){this.uploadFileToUrl=function(o,l,i){var t=new FormData;t.append("file",o),e.post(l,t,{transformRequest:angular.identity,headers:{"Content-Type":void 0}}).success(function(e){i&&i()}).error(function(e){})}}]);
```
Comment (abhilashsajeev) [inline public/source-integrated-file.js:1]: Do not commit generated files

File: routes/index.js [hunk 20]
```diff
@@ -1,12 +1,25 @@
...
+var upload = multer({
+  dest: k.UPLOAD_FOLDER_PATH
+});
 
 /* GET home page. */
 router.get('/', function (req, res, next) {
-  res.render('index', {
-    title: 'Express'
-  });
+  var filePath = path.resolve(__dirname + '/../public/index.html');
```
Comment (abhilashsajeev) [inline routes/index.js:13]: use config to manage location

File: public/file-upload/file-upload.html [hunk 21]
```diff
@@ -0,0 +1,12 @@
+<div class="container text-center div_Modification">
+  <h1>S3 Uploader</h1>
+  <p class="container text-center">
+  	<input type="file" file-model="myFile" accept=".xlsx">
+  </p>
+  <p>
+    <button class="btn btn-default btn-primary" ng-click="uploadFile()" ng-disabled="!myFile">upload me</button>	
+	</p>
+	<div class="alert alert-success" role="alert" ng-show="alertShow">
+  <a href="#" class="alert-link">File uploaded successfully!!</a>
```
Comment (abhilashsajeev) [inline public/file-upload/file-upload.html:10]: Is there any error messages?

**Comment by abhilashsajeev** [comment]
merging for now fix this in later commits

**Pull request merged**